### PR TITLE
DOC: Fix broken reference to count_nonzero in See Also.

### LINKED ISSUE #18525: DOC: sparse doc build warning causing failure (including in CI)
> /home/circleci/repo/build-install/lib/python3.9/site-packages/scipy/sparse/_matrix.py:docstring of scipy.sparse._matrix.spmatrix.getnnz:16: WARNING: py:obj reference target not found: count_nonzero

I bet it is something that has changed with all the `sparse` activity happening at the sprint in Seattle.

### PR DESCRIPTION
**Pull request by rossbar**
This should fix the broken reference for the doc build.

Fixes #18525

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -166,6 +166,7 @@
         ):
     warnings.filterwarnings(
         'once', message='.*' + key)
+warnings.filterwarnings("ignore", category=DeprecationWarning)
```
Comment (j-bowhay) [inline doc/source/conf.py:169]: Is this needed? Typically we add deprecations here: https://github.com/scipy/scipy/blob/aea3e2a8ef68713eb42d77ce00da2d98e34cc763/doc/source/conf.py#L316-L326

Comment (rossbar) [reply doc/source/conf.py:169]: Indeed, nice catch. That's the result of an accidental `commit -a`: I couldn't figure out how to remove the sphinx elevate-warnings-to-errors option with the `dev.py doc` interface, so I had to modify the config file. Will back it out...

**Comment by j-bowhay** [comment]
I ran the doc build locally on this branch and am still getting an error:
```
WARNING: autodoc: failed to import method 'spmatrix.astype' from module 'scipy.sparse'; the following exception was raised:
Traceback (most recent call last):
  File "/home/jakeb/miniconda3/envs/scipy-dev/lib/python3.10/site-packages/sphinx/util/inspect.py", line 376, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'spmatrix' has no attribute 'astype'
```

**Comment by rossbar** [comment]
> I ran the doc build locally on this branch and am still getting an error:

I can't reproduce this locally nor do I see it in the CI logs post 9615583. FWIW I installed the doc dependencies with `pip install -r doc_requirements.txt`. I'm happy to share the results of `pip list` if you think it'd be useful, just LMK!

**Comment by h-vetinari** [comment]
> I ran the doc build locally on this branch and am still getting an error:

That's could be a follow-on investigation. As it stands, this fixes the doc build here, so I'm putting it in. Thanks a lot @rossbar!

**Review by h-vetinari — APPROVED** [review]

**Pull request merged**